lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -1033,6 +1021,14 @@ const char16* WasmBinaryReader::ReadInlineName(uint32& length, uint32& nameLengt
         AssertMsg(UNREACHED, "We calculated the length before decoding, what happened ?");
         ThrowDecodingError(_u("Error while decoding utf8 string"));
     }
+    for (size_t i = 0; i < decodedLength; ++i)
+    {
+        const char16 c = contents[i];
+        if (!utf8::IsValidWideChar(c) || c == UNICODE_UNKNOWN_CHAR_MARK)
+        {
+            ThrowDecodingError(_u("Invalid UTF-8 encoding"));
```

Comment:
Does this work as intended (and as the error message says) with the whole range of Unicode strings? Additionally, should utf8::DecodeUnitsIntoAndNullTerminate be able to generate invalid char16s? Additionally, is this sort of twobyte-by-twobyte validation possible? The docs seem to imply no, as the official validation involves surrogate pairs.